packages/braid-design-system/src/lib/components/Textarea/formatRanges.test.ts

```diff
@@ -180,24 +202,114 @@ describe('formatRanges', () => {
       { start: 30, end: 34 },
       { start: 40, end: 44 },
     ];
-    expect(formatRanges(value, ranges)).toMatchInlineSnapshot(`
+    expect(formatRanges(value, ranges, 'critical')).toMatchInlineSnapshot(`
       [
-        <Highlight>
+        <Highlight
+          tone="critical"
+        >
           some
         </Highlight>,
         " text ",
-        <Highlight>
+        <Highlight
+          tone="critical"
+        >
           some text some text so
         </Highlight>,
-        <Highlight>
+        <Highlight
+          tone="critical"
+        >
           me
         </Highlight>,
         " text ",
-        <Highlight>
+        <Highlight
+          tone="critical"
+        >
           some
         </Highlight>,
         " text",
       ]
     `);
   });
+
+  it('should highlight ranges with caution when tone is caution', () => {
+    const value = 'my longer text';
+    const ranges = [
+      {
+        start: 3,
+        end: 6,
+      },
+      {
+        start: 7,
+        end: 8,
+      },
+      {
+        start: 12,
+        end: 13,
+      },
+    ];
+    expect(formatRanges(value, ranges, 'caution')).toMatchInlineSnapshot(`
+      [
+        "my ",
+        <Highlight
+          tone="caution"
+        >
+          lon
+        </Highlight>,
+        "g",
+        <Highlight
+          tone="caution"
+        >
+          e
+        </Highlight>,
+        "r te",
+        <Highlight
+          tone="caution"
+        >
+          x
+        </Highlight>,
+        "t",
+      ]
+    `);
+  });
+
+  it('should highlight ranges with critical when tone is explicitly critical', () => {
```

Comment:
Are these tests relevant any more now that `formatRanges` requires a `tone`?

Reply:
Yeah, now that it requires a tone, i can remove the defaulting test cases.